### ISSUE TITLE
repeated core dumps doing ord index on 0.5.0 on 16gb ram t2.xlarge instance

### ISSUE THREAD

**Issue by AdrianScott** [ordinals/ord#1712]
experiencing repeated core dumps

ord index running fine from local machine w/ 64gb ram so far

both ubuntu 22

**Comment by andrewtoth** [comment]
Would you be willing to test my speedup branch? It contains a bunch of patches to improve sync speed. Notably skipping blocks before the first inscription height, using p2p interface to fetch blocks and rest interface to fetch txs instead of RPC. Instructions in https://github.com/casey/ord/issues/1648#issuecomment-1426565987, and there's more context in that issue.

**Comment by AdrianScott** [comment]
Sounds neat, great moving re skipping pre-inscription blocks, was thinking about that. I'll take a look at the details, thanks.

**Comment by casey** [comment]
Can someone confirm whether this is a new issue with the 0.5.0 release, or an old issue? Nailing down the change that introduced this problem, if it's a new issue, is highest priority here, since east fix to unblock users is simply to revert whatever that change was.

@cberner, author of redb, not sure if this is redb related, but pinging you just in case it is.

**Comment by AdrianScott** [comment]
This is a new issue for me with the 0.5.0 release. I previously indexed using 0.4.0 on the same instance and have kept a backup of the previous index directory. Unlike my local dev machine, I don't have rust installed on this aws instance (so not rushing to try andrew's solution just yet). Thinking of sftp-ing up local dev machine's index which is almost done now.

**Comment by cberner** [comment]
What error message do you get? There's this open issue in redb: https://github.com/cberner/redb/issues/519

**Comment by AdrianScott** [comment]
Bus error (core dumped)

**Comment by cberner** [comment]
I'd suggest patching ord to use `create()` / `open()` (https://docs.rs/redb/latest/redb/struct.Builder.html#method.create) instead of `create_mmapped()`. If that solves it, then this is most likely an issue with the mmap management in redb. The mmap code path relies on a ton of unsafe code, which I'm working on reducing as part of https://github.com/cberner/redb/issues/458

**Comment by AdrianScott** [comment]
Great, thanks, my local machine completed indexing, so I will upload that index. I'll explore trying that option out or andrew's code the next time i hit the issue, or when I have some time to try it out.

**Comment by AdrianScott** [comment]
Ok, good news, I think I have found the issue & it may affect others.

Many server instances e.g. AWS create the primary / OS volume at 8-10GB.

index.redb for ord 0.4 file size is approximately 1.5GB.

For 0.5, the index.redb is approximately 43GB as I write this.

Default location is in ~/.local/share, i.e. on primary volume on many cloud server instances.

So primary volume fills up and ord core dumps.

I suggest changing the default location of the index.redb, given the new larger size I'm seeing, or this will happen to many people.

I moved the dir and soft-linked to the new location and it looks like the indexing is proceeding without any problems.

Thanks for the help with debugging.
